File: src/main/java/com/blogly/blogly/domain/Title.java
package com.blogly.blogly.domain;

public record Title(String value) {

    private static final int MAX_LENGTH = 255;

    public Title {
        if (value == null || value.isBlank()) {
            throw new IllegalArgumentException("Title value cannot be null or blank");
        }

        if (value.length() > MAX_LENGTH) {
            throw new IllegalArgumentException("Title value cannot exceed " + MAX_LENGTH + " characters");
        }
    }
}

File: src/main/java/com/blogly/blogly/domain/post/Title.kt
package com.blogly.blogly.domain.post

@JvmInline
value class Title(val value: String) {
    init {
        require(value.isNotBlank()) { "Title cannot be blank" }
        require(value.length <= MAX_LENGTH) { "Title cannot exceed $MAX_LENGTH characters" }
    }

    companion object {
        private const val MAX_LENGTH = 255
    }
}
